stdpopsim/models.py

```diff
@@ -180,36 +181,92 @@ def register_qc(self, qc_model):
             raise ValueError(f"QC model already registered for {self.id}.")
         self.qc_model = qc_model
 
+    def get_sample_sets(self, individuals_per_population, ploidy=2):
+        """
+        Returns a list of `msprime.SampleSet` objects, using the number of
+        individuals sampled from each population as specified in
+        `individuals_per_population`.  For instance,
+        ``model.get_sample_sets({"pop_0":5, "pop_1":5}`` would return sample
+        sets encompassing 10 individuals, equally split between populations
+        named "pop_0" and "pop_1".  Omitted populations are assumed to have
+        zero samples.
+
+        :param dict individuals_per_population: A dict of the form
+            `{population_name:number_of_samples}` giving sample counts per
+            population.
+        :param int ploidy: The ploidy of the samples.
+
+        .. note: This interface is intended for internal use only.
+        """
+
+        sample_sets = []
+        population_indices = {x.name: i for i, x in enumerate(self.populations)}
```

Comment:
Slightly simpler to do this I think:
```python
pop_map = {pop.name: pop for pop in self.populations}
for name, num_samples in individuals_per_population.items():
      if name not in pop_map: 
                ....
      pop = pop_map[name]
      if pop.allow_samples:
           ...
```
You can use the ``name`` as the population argument for SampleSet.

Reply:
It turns out that other parts of the code expect integers for the population attribute of SampleSet -- so I'm keeping the use of population indices rather than population names.